test-data/unit/check-deprecated.test

```diff
@@ -377,6 +377,171 @@ for i in a:  # E: function __main__.A.__iter__ is deprecated: no iteration
 [builtins fixtures/tuple.pyi]
 
 
+[case testDeprecatedOverloadedInstanceMethods]
+# flags: --enable-error-code=deprecated
+
+from typing import Iterator, Union
+from typing_extensions import deprecated, overload
+
+class A:
+    @overload
+    @deprecated("pass `str` instead")
+    def f(self, v: int) -> None: ...
+    @overload
+    def f(self, v: str) -> None: ...
+    def f(self, v: Union[int, str]) -> None: ...
+
+    @overload
+    def g(self, v: int) -> None: ...
+    @overload
+    @deprecated("pass `int` instead")
+    def g(self, v: str) -> None: ...
+    def g(self, v: Union[int, str]) -> None: ...
+
+    @overload
+    def h(self, v: int) -> A: ...
+    @overload
+    def h(self, v: str) -> A: ...
+    @deprecated("use `h2`  instead")
+    def h(self, v: Union[int, str]) -> A: ...
+
+class B(A): ...
+
+a = A()
+a.f(1)  # E: overload def (self: __main__.A, v: builtins.int) of function __main__.A.f is deprecated: pass `str` instead
+a.f("x")
```

Comment:
I would also test this case:
```suggestion
a.f("x")

int_or_str: Union[int, str]
a.f(int_or_str)
```

It should not raise if all is good.

Reply:
It does not raise a warning, but why do you think it should not? 

(There is not even a warning for `a.h`, where the implementation is marked as deprecated, which is inconsistent with how functions are handled.)

Reply:
> (There is not even a warning for a.h, where the implementation is marked as deprecated, which is inconsistent with how functions are handled.)

I made it consistent in c95f93678317430e83c256fdd52d578b207eb065.  However, I added the in my opinion missing warnings to the test case, to prevent us from merging this too early by accident.  I am curious to hear why you think the current behaviour is correct. (regarding `int_or_str`).